### PR TITLE
FFWEB-2168: Add Category export

### PR DESCRIPTION
**Pull request by paulfcdd**
- Fixes:
FFWEB-2168

- Description: 
Add Category export

- Tested with Oxid EShop editions/versions: 
6.2.4

- Tested with PHP versions: 
7.4.20

### CODE REVIEW

File: src/Export/Field/BaseFieldInterface.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\FactFinder\Oxid\Export\Field;
+
+interface BaseFieldInterface
```
Comment (a-laurowski) [inline src/Export/Field/BaseFieldInterface.php:7]: I find this confusing to be honest. We have this empty interface and FieldInterface which inherits from it but in some places we have 
`use Omikron\FactFinder\Oxid\Export\Field\BaseFieldInterface as FieldInterface`
i think alias name should not be equal to the class or interface name when what we are aliasing is actually extending it.

What is the benefit of introducing BaseFieldInterface?

Comment (paulfcdd) [reply src/Export/Field/BaseFieldInterface.php:7]: Exporter use field classes (like Omikron\FactFinder\Oxid\Export\Field\Brand), which are implements FieldInterface, I thought that maybe in future we will need similar mechanism for field of Categories or any other export types. So I decide to create common interface for all existing and possible field types.

Comment (a-laurowski) [reply src/Export/Field/BaseFieldInterface.php:7]: Why FieldInterface could not be such a common Interface?

Comment (paulfcdd) [reply src/Export/Field/BaseFieldInterface.php:7]: yeah, in general you're right, let me change it

**Review by a-laurowski — APPROVED** [review]

**Pull request merged**